app/views/granules/download.sh.erb

```diff
@@ -1,19 +1,78 @@
 #!/bin/sh
 
-if command -v curl >/dev/null 2>&1
-then
-    fetch_urls() {
-        while read -r line; do curl -Og $line; done;
-    }
-
-elif command -v wget >/dev/null 2>&1
-then
-    fetch_urls() {
-        while read -r line; do wget $line; done;
-    }
-else
-    fail_with_error "Error: Could not find a command-line downloader.  Please install curl or wget"
-fi
+GREP_OPTIONS=''
+
+cookiejar=$(mktemp cookies.XXXXXXXXXX)
+netrc=$(mktemp netrc.XXXXXXXXXX)
+chmod 0600 "$cookiejar" "$netrc"
+function finish {
+  rm -rf "$cookiejar" "$netrc"
+  <% if @urls.first.start_with? 'ftp' %>
+  <% # For FTP retrievals, we also need to clean up ~/.netrc file for safety reasons %>
+  dot_netrc_tmp=$(mktemp ~/.netrc.tmp.XXXXXXXXXX)
+  chmod 0600 "$dot_netrc_tmp"
+  awk '!/<%= @urls.first[/ftp:\/\/([^\/]*)\/.*/, 1].gsub('.', '\.') %>/' ~/.netrc > $dot_netrc_tmp && mv $dot_netrc_tmp ~/.netrc
+  <% end %>
+}
+
+trap finish EXIT
+WGETRC="$wgetrc"
+
+prompt_credentials() {
+    echo "Enter your Earthdata Login credentials"
+    read -p "Username (<%= @user %>): " username
+    username=${username:-<%= @user %>}
+    read -s -p "Password: " password
+    echo "\nmachine urs.earthdata.nasa.gov\tlogin $username\tpassword $password" >> $netrc
+    <% if @urls.first.start_with? 'ftp' %>
+    echo "\nmachine <%= @urls.first[/ftp:\/\/([^\/]*)\/.*/, 1] %>\tlogin $username\tpassword $password" >> $netrc
+    <% end %>
+    echo
+}
+
+exit_bad_login() {
+    echo
+    echo "Unable to log in"
+    echo
+    echo $1
+    echo
+    echo "<%=raw @urls.first %>"
```

Comment:
```$1``` is the passed-in exit reason. We show user two error messages on exiting. Is this what you are referring to?